server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -752,11 +708,15 @@ private static void parseDynamicValue(final ParseContext context, ObjectMapper p
         if (dynamic == ObjectMapper.Dynamic.FALSE) {
             return;
         }
-        final Mapper.Builder builder = createBuilderFromDynamicValue(context, token, currentFieldName);
-        Mapper mapper = builder.build(context.path());
-        context.addDynamicMapper(mapper);
-
-        parseObjectOrField(context, mapper);
+        DynamicFieldsBuilder dynamicFieldsBuilder = DynamicFieldsBuilder.forDynamic(dynamic);
+        DynamicField dynamicField = createDynamicFieldFromValue(context, token, currentFieldName, dynamicFieldsBuilder);
+        if (dynamicField.isRuntimeField()) {
```

Comment:
Maybe the builders should call these methods on the context?

Reply:
can you expand? I don't follow

Reply:
Like, instead of returning the runtime field or whatever it can just call `addDynamicRuntimeField` if it is making a runtime field or `addDynamicMapper` if it is making a dynamic mapper.

Reply:
ah I see, this makes sense, I will try this out.